src/mutations/bookings.ts

```diff
@@ -13,6 +14,12 @@ const createBooking = async (parent, data, ctx): Promise<Booking> => {
   const room: Room = await ctx.prisma.room({
     number: data.room_number,
   });
+
+  const existingBookings = await ctx.prisma
+    .room({ number: data.room_number })
+    .bookings({ orderBy: "start_ASC" });
```

Comment:
Optimizing this query will make a lot of performance improvements. Consider doing something like this:
```
const existingBookings = await ctx.prisma
    .room({ number: data.room_number })
    .bookings({ 
        orderBy: "start_ASC",
        where: {
          start_gte: <DateTime>
          end_lte: <DateTime>
        }
});
```

Reply:
I'll make this change

Reply:
damn it, this would actually solved the entire problem
could just do
```
where: {
  start_gte: start,
  end_lte: end,
}
```
if `existingBookings.length > 0` it means there should be a collision

Reply:
no this doesn't solve for all cases